src/Rec/Rec.ts

```diff
@@ -0,0 +1,467 @@
+ * predicate function returns `true`.
+ *
+ * @group Transformations
+ * @group Filtering
+ */
+export const filter =
+    <K extends string, V>(f: (k: NoInfer<K>, v: NoInfer<V>) => boolean) =>
+    (rec: Rec<K, V>): Rec<K, V> => {
+        if (isEmpty(rec)) {
+            return rec
+        }
+
+        const out: Record<K, V> = empty<K, V>()
+
+        for (const k in rec) {
+            if (f(k, rec[k])) {
+                out[k] = rec[k]
+            }
+        }
+
+        return out
+    }
+
+/**
+ * Test whether every key/value pair in a record returns `true` for the
+ * given predicate function.
+ *
+ * @group Utils
+ */
+export const every =
+    <K extends string, V>(f: (k: NoInfer<K>, v: NoInfer<V>) => boolean) =>
+    (rec: Rec<K, V>): boolean => {
+        if (isEmpty(rec)) {
+            return true
+        }
+
+        for (const k in rec) {
+            if (!f(k, rec[k])) {
+                return false
+            }
+        }
+
+        return true
+    }
+
+/**
+ * Execute an arbitrary side-effect function for every key/value pair in the record.
+ * Does not affect the values contained in the map. Can be helpful for logging
+ * or debugging.
+ *
+ * @group Utils
+ *
+ * @param f Should not mutate its arguments. See {@link map} if you want to
+ * transform the record into a new record with different values.
+ *
+ * @returns void
+ */
+export const iter =
+    <K extends string, V>(f: (k: NoInfer<K>, v: NoInfer<V>) => void) =>
+    (rec: Rec<K, V>): void => {
+        if (isEmpty(rec)) {
+            return
+        }
+
+        for (const k in rec) {
+            f(k, rec[k])
+        }
+    }
+
+/**
+ * Convert an array of tuples (key-value pairs) into a `Rec`.
+ *
+ * @group Constructors
+ */
+export const ofArray = <K extends string, V>(
+    array: readonly (readonly [K, V])[],
+    equalityComparer?: EqualityComparer<K>
+): Rec<K, V> => {
+    if (array.length < 1) {
+        return empty()
+    }
+
+    return array.reduce<Rec<K, V>>(
+        (rec, kvp) => pipe(rec, set(kvp, equalityComparer)),
+        empty()
+    )
+}
+
+/**
+ * Get a new record with the given key removed. For key lookup, will use
+ * the `EqualityComparer` if passed, otherwise defaults to reference equality
+ * (triple equals). The record will be returned unchanged if the key is not
+ * found.
+ *
+ * @group Transformations
+ */
+export const remove =
+    <K extends string>(
+        key: NoInfer<K>,
+        equalityComparer?: EqualityComparer<NoInfer<K>>
+    ) =>
+    <V>(rec: Rec<K, V>) =>
+        pipe(
+            rec,
+            findWithKey(key, equalityComparer),
+            Option.match({
+                some: ([k, _]) => {
+                    const copy = { ...rec }
+                    delete copy[k]
+                    return copy
+                },
+                none: rec,
+            })
+        )
+
+/**
+ * Convert a `Record` object into a `Rec`. Useful if you need to use a _more restrictive_
```

Comment:
more or less restrictive? The example below sounds _less_ restrictive. Or am I misunderstanding?

Reply:
Ah that's a good point. I think I should just say "different" lol

Reply:
https://github.com/fp-toolkit/fp-toolkit/pull/67/commits/09f3f5c565a1e8986b5b05dce808421d2981dacf